input/drivers/android_input.c

```diff
@@ -869,7 +871,7 @@ static void handle_hotplug(android_input_data_t *android_data,
    if (!back_mapped && settings->input.back_as_menu_toggle_enable)
       settings->input.autoconf_binds[*port][RARCH_MENU_TOGGLE].joykey = AKEYCODE_BACK;
 
-   android_data->pad_states[android_data->pads_connected].id = id;
+   android_data->pad_states[*port].id = id;
```

Comment:
why are you changing this? this will affect other implementations and it was changed for a reason